backends/ceed-backend-weak.c

```diff
@@ -1,17 +1,30 @@
 #include <ceed/backend.h>
+#include <limits.h>
 #include <stdlib.h>
 #include <stdio.h>
 
-// This function provides a debug target for weak symbols
 // LCOV_EXCL_START
-static int CeedRegister_Weak(const char *name) {
+// This function provides improved error messages for uncompiled backends
+static int CeedInit_Weak(const char *resource, Ceed ceed) {
+  return CeedError(ceed, CEED_ERROR_UNSUPPORTED,
+                   "Backend not currently compiled: %s", resource);
+}
+
+// This function provides a debug target for weak symbols
+static int CeedRegister_Weak(const char *name, int num_prefixes, ...) {
   if (getenv("CEED_DEBUG")) fprintf(stderr, "Weak %s\n", name);
+  va_list prefixes;
+  va_start(prefixes, num_prefixes);
+  for (int i=0; i<num_prefixes; i++) {
+    CeedRegister(va_arg(prefixes, const char*), CeedInit_Weak, UINT_MAX);
```

Comment:
Registered with largest possible priority value so that `CeedInit_Weak` only runs on an exact character for character match